src/nvim/api/vim.c

```diff
@@ -72,10 +73,70 @@ void api_vim_free_all_mem(void)
   map_free(String, handle_T)(namespace_ids);
 }
 
+/// Executes Vimscript (multiline block of Ex-commands), like anonymous
+/// |:source|.
+///
+/// Optionally returns (non-error, non-shell |:!|) output.
+///
+/// On execution error: fails with VimL error, does not update v:errmsg.
+///
+/// @see |execute()|
+/// @see |nvim_command()|
+///
+/// @param src      Vimscript code
+/// @param output   Capture and return all (non-error, non-shell |:!|) output
+/// @param[out] err Error details (Vim error), if any
+String nvim_exec(String src, Boolean output, Error *err)
```

Comment:
this function is almost identical to `nvim_command_output`. Try to avoid duplicating lots of code. (But, I will do so in this case)